test/integration/resends-cancelled-on-client-disconnect.test.ts

```diff
@@ -95,16 +137,29 @@ describe('resend cancellation', () => {
         await tracker.stop()
     })
 
-    it('on client disconnect: associated resend is cancelled', (done) => {
+    beforeAll(async () => {
+        mockDataQueryServer = await createMockDataServer()
+    })
+
+    afterAll(async () => {
+        mockDataQueryServer.close()
+        await once(mockDataQueryServer, 'close')
+    })
+
+    it('on client disconnect: associated resend is cancelled', async () => {
```

Comment:
I'm left wondering if there could be a more straightforward way to test the same with less mocking & setup required. Anyways, I could use a walk though of this test in call or smth.